### PR TITLE
WebGPU: fix sparse color attachments/formats binding

### PR DESCRIPTION
**Pull request by shrekshao**

### CODE REVIEW

File: src/library_webgpu.js [hunk 1]
```diff
@@ -1254,6 +1255,7 @@ var LibraryWebGPU = {
     }
 
     function makeColorState(csPtr) {
+      if (!csPtr) return undefined;
```
Comment (kainino0x) [inline src/library_webgpu.js:1258]: the `csPtr` won't be null, it's just a pointer inside an array (see how it's invoked in `makeColorStates`). Instead we need to return `undefined` if `WGPUColorTargetState.format` is 0.

**Comment by shrekshao** [comment]
Thanks for catching these (face_palm)

**Review by kainino0x — APPROVED** [review]

**Comment by kainino0x** [comment]
LGTM, not sure what the failure is, might need rebase and retry (or maybe build is just broken at the moment)

**Comment by kripken** [comment]
(Not sure what those CI errors are, but yes, merging in latest `main` should help - not aware of general issues atm)

**Comment by shrekshao** [comment]
Rebased on main and now only failing test_nodejs_sockets_echo_tcp. https://app.circleci.com/pipelines/github/emscripten-core/emscripten/21999/workflows/f3532467-cca9-4568-ac6d-d49ad03f53a9/jobs/544728
Seems unrelated. I will try to trigger a rerun (by force push)

**Pull request merged**